deepchem/data/datasets.py

```diff
@@ -2269,39 +2237,20 @@ def make_pytorch_dataset(self, epochs: int = 1, deterministic: bool = False):
     `torch.utils.data.IterableDataset` iterating over the same data as
     this dataset.
     """
-    import torch
-
-    def get_image(array, index):
-      if isinstance(array, np.ndarray):
-        return array[index]
-      return dc.data.ImageLoader.load_img([array[index]])[0]
-
-    def iterate():
-      n_samples = self._X_shape[0]
-      worker_info = torch.utils.data.get_worker_info()
-      if worker_info is None:
-        first_sample = 0
-        last_sample = n_samples
-      else:
-        first_sample = worker_info.id * n_samples // worker_info.num_workers
-        last_sample = (
-            worker_info.id + 1) * n_samples // worker_info.num_workers
-      for epoch in range(epochs):
-        if deterministic:
-          order = first_sample + np.arange(last_sample - first_sample)
-        else:
-          order = first_sample + np.random.permutation(last_sample -
-                                                       first_sample)
-        for i in order:
-          yield (get_image(self._X, i), get_image(self._y, i), self._w[i],
-                 self._ids[i])
-
-    class TorchDataset(torch.utils.data.IterableDataset):  # type: ignore
-
-      def __iter__(self):
-        return iterate()
-
-    return TorchDataset()
+    try:
+      from deepchem.data.pytorch_datasets import TorchImageDataset
+    except:
+      raise ValueError("This method requires PyTorch to be installed.")
+
+    pytorch_ds = TorchImageDataset(
+        X=self.X,
+        y=self.y,
+        w=self.w,
```

Comment:
I think you meant this to be `self._X`, `self._y`, and `self._w`?  What you've written here will force it to load all images into memory immediately.

Reply:
It also might be cleaner to just pass a reference to the ImageDataset, the same way you do for DiskDataset, rather than passing all its fields individually.  Same for NumpyDataset.

Reply:
I applied the same way I did for DiskDataset to TorchNumpy(Image)Dataset